jparest/src/test/java/uk/gov/homeoffice/digital/sas/jparest/controller/ResourceApiControllerTest.java

```diff
@@ -582,6 +583,309 @@ void update_idExistsOnRequestPathAndDoesNotExistOnBody_verifyPayloadIsValidatedW
+            DESCRIPTION_FIELD_NAME, "Updated Dummy Entity C Two",
+            INDEX_FIELD_NAME, 2, TENANT_ID_FIELD_NAME, TENANT_ID));
+
+        var updatedPayload = new String[] {updatedResourceOnePayload, updatedResourceTwoPayload};
+
+        var payload = Arrays.toString(updatedPayload);
+
+        var updateResponse = controller.batchUpdate(TENANT_ID, payload);
+
+
+        assertThat(updateResponse.getItems()).hasSize(2);
+
+        var dummyOne = updateResponse.getItems().get(0);
+        assertThat(dummyOne).isNotNull();
+        assertThat(dummyOne.getId()).isEqualTo(createdResource.getId());
+        assertThat(dummyOne.getIndex()).isEqualTo(2);
+        assertThat(dummyOne.getDescription()).isEqualTo("Updated Dummy Entity C One");
+
+        var dummyTwo = updateResponse.getItems().get(1);
+        assertThat(dummyTwo).isNotNull();
+        assertThat(dummyTwo.getId()).isEqualTo(createdResource2.getId());
+        assertThat(dummyTwo.getIndex()).isEqualTo(2);
+        assertThat(dummyTwo.getDescription()).isEqualTo("Updated Dummy Entity C Two");
+
+        var checkResponse = controller.get(TENANT_ID, createdResource.getId());
+        var checkResource = checkResponse.getItems().get(0);
+        assertThat(checkResource).isEqualTo(dummyOne);
+
+        var checkResponse2 = controller.get(TENANT_ID, createdResource2.getId());
+        var checkResource2 = checkResponse2.getItems().get(0);
+        assertThat(checkResource2).isEqualTo(dummyTwo);
+    }
+
+    @ParameterizedTest(name="{0}")
+    @MethodSource("invalidPayloads")
+    @Transactional
+    void batchUpdate_resourceExistsInvalidPayload_jsonExceptionThrown(String payload) {
+        var controller = getResourceApiController(DummyEntityA.class);
+        assertThatExceptionOfType(JsonProcessingException.class).isThrownBy(() -> controller.batchUpdate(TENANT_ID, payload));
+    }
+
+    @ParameterizedTest(name="{0}")
+    @MethodSource("invalidPayloads")
+    @Transactional
+    void batchUpdate_resourceDoesntExistInvalidPayload_jsonExceptionThrown(String payload) {
+
+        var controller = getResourceApiController(DummyEntityA.class);
+
+        assertThatExceptionOfType(ResourceNotFoundException.class).isThrownBy(() -> controller.get(TENANT_ID, NON_EXISTENT_ID));
+        assertThatExceptionOfType(JsonProcessingException.class).isThrownBy(() -> controller.batchUpdate(TENANT_ID, payload));
+    }
+
+    @ParameterizedTest(name="{0}")
+    @MethodSource("invalidBatchProperty")
+    void batchUpdate_unrecognizedPropertyOnPayload_unknownResourcePropertyExceptionThrown(String payload) {
+        var controller = getResourceApiController(DummyEntityA.class);
+        assertThatExceptionOfType(UnknownResourcePropertyException.class).isThrownBy(() -> controller.batchUpdate(TENANT_ID, payload));
+    }
+
+    @Test
+    @Transactional
+    void updateMultiple_oneResourceDoesntExist_resourceNotFoundExceptionThrown() throws JsonProcessingException {
```

Comment:
nit: should this say `batchUpdate` instead of `updateMultiple`?

Reply:
yep, I missed that one! Updated